File: frontend/src/lib/api/sync.ts
import apiClient from "./client";

/**
 * POST /sync/trigger request body. `job_types` is optional: when omitted the
 * backend picks a platform-appropriate default set (see backend
 * services.sync.DEFAULT_JOB_TYPES). Only job_types with a producer for the
 * account's platform are dispatched.
 */
export interface TriggerSyncRequest {
  account_id: string;
  job_types?: string[];
}

export interface TriggerSyncResponse {
  message: string;
  job_ids: string[];
}

export const syncApi = {
  status: (accountId: string) =>
    apiClient.get("/sync/status", { params: { account_id: accountId } }),
  trigger: (accountId: string, jobTypes?: string[]) => {
    const body: TriggerSyncRequest = { account_id: accountId };
    if (jobTypes !== undefined) body.job_types = jobTypes;
    return apiClient.post("/sync/trigger", body);
  },
};

File: backend/app/schemas/sync.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class JobStatus(BaseModel):
    status: str
    last_run_at: Optional[datetime] = None
    next_run_at: Optional[datetime] = None
    is_stale: bool = False
    percent_complete: Optional[int] = None


class RateLimitStatus(BaseModel):
    insights_app_pct: float = 0
    insights_acc_pct: float = 0
    structure_acc_pct: float = 0
    access_tier: str = "standard_access"


class SyncStatusResponse(BaseModel):
    account_id: str
    jobs: dict[str, JobStatus]
    rate_limit: RateLimitStatus
    has_warning: bool


class TriggerSyncRequest(BaseModel):
    account_id: str
    # Optional: when omitted, the service picks a platform-appropriate default
    # set (see services.sync.DEFAULT_JOB_TYPES). Only job_types that have a
    # producer for the account's platform are dispatched.
    job_types: Optional[list[str]] = None


class TriggerSyncResponse(BaseModel):
    message: str
    job_ids: list[str]
